.github/workflows/push-deploy-website.yml

```
@@ -0,0 +1,64 @@
+          node-version: 22
+          cache: pnpm
+          cache-dependency-path: 'docs/'
+      - uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.DEPLOY_GITHUB_PAGES }}
+      - name: Deploy to GitHub Pages
+        env:
+          USE_SSH: true
+          GIT_USER: git
+        working-directory: ./docs
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2181:style:4:6: Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?

(shellcheck)